bazel/re2c.bzl

```diff
@@ -0,0 +1,81 @@
+# re2c generator rules - assuming that re2c is available as '@re2c//:re2c'.
+
+visibility("public")
+
+def _re2c_impl(ctx):
+    srcs = ctx.files.srcs
+    if len(srcs) != 1:
+        fail("Attribute `srcs` must contain exactly 1 '.re2c' file.")
+    outs = ctx.outputs.outs
+    if not outs or [o for o in outs if o.path.split(".")[-1] not in ["c", "cc", "h"]]:
+        fail("Attribute `outs` must contain at least 1 '.cc' or '.h' file.")
```

Comment:
Should we also add .hpp and .cpp, perhaps even .cxx?

Also, isn't there a standard function for taking file extension?